packages/data-sdk/src/utils/RtcClientPool.ts

```diff
@@ -0,0 +1,88 @@
+import { RtcClient, IRtcClientConfiguration } from "@formant/realtime-sdk";
+
+export interface PooledRtcClient extends RtcClient {
+  release(): Promise<boolean>;
+}
+
+type ReceiveFn = IRtcClientConfiguration["receive"];
+type CreateClientFn = (receive: ReceiveFn) => RtcClient;
+
+export interface IRtcClientPoolOptions {
+  createClient: CreateClientFn;
+}
+
+const notAllowed = (err: string) => () => {
+  throw new Error(err);
+};
```

Comment:
I've never seen this pattern before. curious about when to use it? it returns a function with the error. Why don't we just return the error.

Reply:
actually, i should probably remove this now. I _previously_ had a few different `notAllowed` errors, and used this as a generator function to make them, but now there is _only one left_ so it probably would be easier to just proc the error in place. 😬